Fix typos and warnings

neelkamath opened the pull request.

Commit:
Rename `bool` to `boolean` as it's Java

addisonElliott commented:
Good catch. I found another instance where `bool` is used instead of `boolean`. It's a few lines above where it says:

> So far we’ve used values with type String, Integer, bool, and ParseObject. Parse also supports float, java.util.Date, and JSONObject.NULL.

Do you mind changing that as well to get them all in one PR?

Commit:
Fix other bool to boolean

Commit:
Merge pull request #1 from parse-community/gh-pages

Update getting-started.md (#513)

Commit:
Update `compile` to `implementation`

`compile` will be deprecated by the end of 2018 and should be changed to `implementation`

Commit:
Rename "option" (incorrect) to "step"

neelkamath commented:
I put that single commit of 'renaming "option" to "step"' from the other PR in this one because that time I was too lazy to commit on the same branch and now it'd also cause merge issues.

flovilmart commented:
Thanks @neelkanath for the PR and @addisonElliott for the review!
